engine/src/main/java/org/apache/apex/stram/DeployManager.java

```diff
@@ -0,0 +1,232 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.apex.stram;
+
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.apex.stram.DeployRequest.EventGroupId;
+
+import com.google.common.collect.Maps;
+
+import com.datatorrent.stram.plan.physical.PTOperator;
+
+/**
+ * This class manages tracking ids of deploy/undeploy for containers and
+ * operators.
+ *
+ */
+public class DeployManager
+{
+  private static DeployManager deployManager = new DeployManager();
+  private Map<String, DeployRequest> deployRequests = Maps.newHashMap();
+
+  public static DeployManager getDeployManagerInstance()
+  {
+    return deployManager;
+  }
+
+  /**
+   * Retruns all available deploy requests with StrAM
+   * @return deployRequests
+   */
+  public Map<String, DeployRequest> getDeployRequests()
+  {
+    return deployRequests;
+  }
+
+  /**
+   * Returns deploy request for container
+   * @param containerId
+   * @return deployRequest
+   */
+  public DeployRequest getDeployRequest(String containerId)
+  {
+    return deployRequests.get(containerId);
+  }
+
+  /**
+   * Returns deploy/undeploy group Id for container
+   * @param containerId
+   * @return groupId <br/>
+   *         <b>Note:</b> groupId 0 indicates and independent event, with no
+   *         group
+   */
+  public EventGroupId getEventGroupIdForContainer(String containerId)
+  {
+    EventGroupId groupId = null;
+    if (deployRequests.get(containerId) != null) {
+      groupId = deployRequests.get(containerId).getEventGroupId();
+    }
+    return groupId;
+  }
+
+  /**
+   * Returns deploy/undeploy group Id for container This could be a new
+   * container allocated during redeploy process
```

Comment:
This could even be called during start of an application while allocating new containers. right?